core/src/main/scala/kafka/utils/ShutdownableThread.scala

```diff
@@ -103,6 +112,6 @@ abstract class ShutdownableThread(val name: String, val isInterruptible: Boolean
   }
 
   def isRunning: Boolean = {
-    shutdownInitiated.getCount() != 0
+    shutdownInitiated.getCount != 0
```

Comment:
Perhaps we can fix this separately, but it's a bit weird that this would still return true after a thread failure. It would be more accurate to rename this to `isShutdownInitiated`. and have a separate method like this:
```scala
def isRunning: Boolean = !isShutdownInitiated && !isShutdownComplete
```
Or something like that...

Reply:
I think let's fix it separately. I tried to make this change but other tests failed and it may take time to weed out why did it fail.
For this PR I tried to make this a little bit more clear by refactoring the internals a bit. Let me know your opinion.

Reply:
Sounds good.